src/main/java/com/blamejared/crafttweaker/api/managers/IRecipeManager.java

```diff
@@ -64,10 +64,7 @@ default void addJSONRecipe(String name, IData data) {
         JsonObject recipeObject = JSON_RECIPE_GSON.fromJson(mapData.toJsonString(), JsonObject.class);
         String recipeTypeKey = getBracketResourceLocation().toString();
         
-        if(recipeObject.has("type")) {
-            if(!recipeObject.get("type").getAsString().equals(recipeTypeKey))
-                throw new IllegalArgumentException("Cannot override recipe type! Given: \"" + recipeObject.get("type").getAsString() + "\", Expected: \"" + recipeTypeKey + "\"");
-        } else {
+        if(!recipeObject.has("type")) {
             recipeObject.addProperty("type", recipeTypeKey);
         }
         IRecipe<?> iRecipe = RecipeManager.deserializeRecipe(new ResourceLocation(CraftTweaker.MODID, name), recipeObject);
```

Comment:
```suggestion
        IRecipe<?> iRecipe = RecipeManager.deserializeRecipe(new ResourceLocation(CraftTweaker.MODID, name), recipeObject);
        if (iRecipe.getType() != getRecipeType()) {
            throw new IllegalArgumentException("The type of JSON Recipe is not matching the recipe type!");
        }
```

@kindlich for now, I think can use it to check if the serialized recipe is matching the recipe type.

Reply:
Added this in the most recent commit with some adjustments to the message.